### PR TITLE
fix: new source and squad writing form issues

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Add the missing title for the new source modal.
- Squad writing must be wrapped in form actions.
- Also fixed the uncentered image on mobile (first screenshot).

Screenshot 2024-04-01 at 10 24 45 PM

Screenshot 2024-04-01 at 10 23 03 PM

https://github.com/dailydotdev/apps/assets/13744167/e3d38d2b-c8fe-482b-8269-ec099b350e36

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-251 #done

### CODE REVIEW

**Comment by sshanzel** [comment]
> In your video the cancel goes to a completely different screen then where the user originated from.
> That feels like bad UX no?

Yeah, noticed that too, but then I checked on prod and that is the current behavior.

**Comment by rebelchris** [comment]
@sshanzel Could you maybe open the first 2 questions for design/product (non blocking, but maybe good as follow up)

File: packages/shared/src/components/squads/Details.tsx [hunk 1]
```diff
@@ -165,7 +173,41 @@ export function SquadDetails({
   };
 
   return (
-    <>
+    <ConditionalWrapper
+      condition={isMobile}
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:177]: Should the condition not be `isMobile && createMode`?

File: packages/webapp/pages/squads/new.tsx [hunk 2]
```diff
@@ -79,22 +71,9 @@ const NewSquad = (): ReactElement => {
...
-          <SquadTitle className="mb-2">Create new Squad</SquadTitle>
-          <SquadSubTitle>
-            Create a group where you can learn and interact privately with other
-            developers around topics that matter to you
-          </SquadSubTitle>
-        </div>
         <SquadDetails
           className="p-8 pt-0"
-          form={form}
+          form={{ public: false }}
```
Comment (rebelchris) [inline packages/webapp/pages/squads/new.tsx:76]: What does this setting do?

**Review by rebelchris — APPROVED** [review]
Non blocking question.
Also note my other comment please

File: packages/webapp/pages/squads/new.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/webapp/pages/squads/new.tsx:76]: Not really sure what happens inside, but I just noticed a local state above and doesn't seem to be changed. Instead I added this inline object over the unnecessary hook.

File: packages/shared/src/components/squads/Details.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/Details.tsx:177]: On Edit mode, there should be a "Save" button.

Comment (rebelchris) [reply packages/shared/src/components/squads/Details.tsx:177]: But still in a form wrapper?

**Comment by sshanzel** [comment]
> @sshanzel Could you maybe open the first 2 questions for design/product (non blocking, but maybe good as follow up)

Not sure what's the motivation in making the Squad creation a popup since we are trying to move to pages, but sure, let me ask them.

**Pull request merged**